packages/@angular/cli/utilities/dynamic-path-parser.ts

```diff
@@ -3,19 +3,26 @@ import * as process from 'process';
 import * as fs from 'fs';
 const stringUtils = require('ember-cli-string-utils');
 
-export function dynamicPathParser(project: any, entityName: string, appConfig: any) {
-  const projectRoot = project.root;
-  const sourceDir = appConfig.root;
+export interface DynamicPathOptions {
+  project: any;
+  entityName: string;
+  appConfig: any;
+  dryRun: boolean;
+}
+
```

Comment:
To allow defaults and improve encapsulation, what about something like:
```
export class DynamicPathOptions {
  projectRoot: string;
  sourceDirectory: string;
  entityName: string;
  dryRun?: boolean = true;
}
```
And then something like this in the function:
```
options = { ...new DynamicPathOptions, ...options };
```